assets/quick-order-list.js

```diff
@@ -154,6 +153,7 @@ class QuickOrderList extends HTMLElement {
         const html = new DOMParser().parseFromString(responseText, 'text/html');
         const sourceQty = html.querySelector(`#${this.quickOrderListId}`);
         this.innerHTML = sourceQty.innerHTML;
+        this.defineInputsAndQuickOrderTable();
```

Comment:
```suggestion
```

Can we add this to connected call back instead? Eventually well want to refactor this and take this function out since its reused in other `js` files like cart and quick add bulk

Reply:
I tried to add it to `connectedCallback()` but it didn't work. The reason is because inside callback we need to run `this.onCartUpdate();` first and this takes time since we call API there. Even if I place `this.defineInputsAndQuickOrderTable();` after `this.onCartUpdate();` it won't wait until the `this.onCartUpdate();`done. The `this.defineInputsAndQuickOrderTable();` will run synchronously.

Reply:
I can try to implement it with promises. Let me try it :)

Reply:
Okay, that's how it looks now with another approach [f18f7b6](https://github.com/Shopify/dawn/pull/3360/commits/f18f7b643372a9dd2755e3353b79c759bfa2f60d)

It seems more complex but probably this approach is more appropriate. Thoughts?

Reply:
Yes this works :) thanks for updating it!